web/src/components/CodeEditor/index.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { useState, useRef, useCallback, Dispatch, SetStateAction, useEffect } from "react";
+import Editor, {useMonaco} from "@monaco-editor/react";
+import {ResourceNodeData, Argument} from "../../interfaces/ResourceNodeData";
```

Comment:
Tu as des imports inutiles 
```suggestion
import React, { Dispatch, SetStateAction } from "react";
import Editor from "@monaco-editor/react";
import {ResourceNodeData} from "../../interfaces/ResourceNodeData";
```